src/Microsoft.AspNetCore.Rewrite/Internal/RedirectToWwwRule.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Rewrite.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal
+{
+    public class RedirectToWwwRule : IRule
+    {
+        public readonly int _statusCode;
+
+        public RedirectToWwwRule(int statusCode)
+        {
+            _statusCode = statusCode;
+        }
+
+        public virtual void ApplyRule(RewriteContext context)
+        {
+            var req = context.HttpContext.Request;
+            if (req.Host.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            if (req.Host.Value.StartsWith("www.", StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            }
+
+            var wwwHost = new HostString($"www.{req.Host.Value}");
```

Comment:
You'll get some pretty strange results if the host is 127.0.0.1:80 or [::1]:80. I don't think those are common enough to worry about though.

Reply:
We may want to consider adding a list of excluded hosts, like https://github.com/aspnet/BasicMiddleware/commit/892cebb27eb1487a10e07d70967e490ecb38bce8#diff-15f19c319158f42c2602a43b53630d87R43.

Reply:
No, if it came up you'd want to exclude IP addresses as a class of host, not exclude individual values.